builtin/providers/aws/resource_aws_ecs_service.go

```diff
@@ -241,21 +241,18 @@ func resourceAwsEcsServiceUpdate(d *schema.ResourceData, meta interface{}) error
 func resourceAwsEcsServiceDelete(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ecsconn
 
+	serviceName := d.Id()
+	clusterName := d.Get("cluster").(string)
+
 	// Check if it's not already gone
 	resp, err := conn.DescribeServices(&ecs.DescribeServicesInput{
-		Services: []*string{aws.String(d.Id())},
-		Cluster:  aws.String(d.Get("cluster").(string)),
+		Services: []*string{aws.String(serviceName)},
+		Cluster:  aws.String(clusterName),
 	})
 	if err != nil {
 		return err
 	}
-
-	if len(resp.Services) == 0 {
-		log.Printf("[DEBUG] ECS Service %q is already gone", d.Id())
-		return nil
-	}
```

Comment:
Any reason for removing this? I'm aware that ECS services typically remain in `INACTIVE` state, so it may look like useless 4 lines of code, but imagine a situation where someone has manually removed the service outside of Terraform. If we won't have this condition here, the code below would cause crash since it's expecting non-empty list.